src/battle_script_commands.c

```diff
@@ -3801,6 +3801,53 @@ void SetMoveEffect(bool32 primary, u32 certain)
                     gBattlescriptCurrInstr = BattleScript_SpikesActivates;
                 }
                 break;
+            case MOVE_EFFECT_TRIPLE_ARROWS:
+                {
+                    u8 randomLowerDefenseChance;
+                    u8 randomFlinchChance;
+
+                    if (GetBattlerAbility(gBattlerAttacker) == ABILITY_SERENE_GRACE)
+                    {
+                        randomLowerDefenseChance = RandomPercentage(RNG_TRIPLE_ARROWS_DEFENSE_DOWN, 100);
+                        randomFlinchChance = RandomPercentage(RNG_TRIPLE_ARROWS_FLINCH, 60);
+                    }
+                    else
+                    {
+                        randomLowerDefenseChance = RandomPercentage(RNG_TRIPLE_ARROWS_DEFENSE_DOWN, 50);
+                        randomFlinchChance = RandomPercentage(RNG_TRIPLE_ARROWS_FLINCH, 30);
+                    }
+
+                    if (randomLowerDefenseChance && randomFlinchChance)
+                    {
+                        BattleScriptPush(gBattlescriptCurrInstr + 1);
+                        gBattlescriptCurrInstr = BattleScript_ReduceDefenseAndFlinch;
+                    }
+                    else
+                    {
+                        if (randomLowerDefenseChance)
+                        {
+                            BattleScriptPush(gBattlescriptCurrInstr + 1);
+                            gBattlescriptCurrInstr = BattleScript_DefDown;
+                        }
+                        else if (randomFlinchChance)
+                        {
+                            if (battlerAbility == ABILITY_INNER_FOCUS)
+                            {
+                                gLastUsedAbility = ABILITY_INNER_FOCUS;
+                                gBattlerAbility = gEffectBattler;
+                                RecordAbilityBattle(gEffectBattler, ABILITY_INNER_FOCUS);
+                                gBattlescriptCurrInstr = BattleScript_FlinchPrevention;
```

Comment:
Again, this isn't needed. There's no battle script for when Flinch as a secondary effect triggers Inner Focus. I've recorded a sample from Scarlet as an example to show what I mean:
[https://streamable.com/vhp2ck](https://streamable.com/vhp2ck)